core/services/workflows/execution_strategy.go

```diff
@@ -0,0 +1,151 @@
+package workflows
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+
+	"github.com/smartcontractkit/libocr/permutation"
+
+	"golang.org/x/crypto/sha3"
+)
+
+type executionStrategy interface {
+	Apply(ctx context.Context, l logger.Logger, cap capabilities.CallbackExecutable, req capabilities.CapabilityRequest) (values.Value, error)
+}
+
+var _ executionStrategy = immediateExecution{}
+
+type immediateExecution struct{}
+
+func (i immediateExecution) Apply(ctx context.Context, lggr logger.Logger, cap capabilities.CallbackExecutable, req capabilities.CapabilityRequest) (values.Value, error) {
+	l, err := capabilities.ExecuteSync(ctx, cap, req)
+	if err != nil {
+		return nil, err
+	}
+
+	// `ExecuteSync` returns a `values.List` even if there was
+	// just one return value. If that is the case, let's unwrap the
+	// single value to make it easier to use in -- for example -- variable interpolation.
+	if len(l.Underlying) > 1 {
+		return l, nil
+	}
+
+	return l.Underlying[0], nil
+}
+
+var _ executionStrategy = delayedExecution{}
+
+type delayedExecution struct {
```

Comment:
delayedExecution might suggest that we are executing with a delay, which is not really the case - one of the nodes executes immediately. How about "scheduledExecution"?